lib/vagrant-cookbook-fetcher/action_set_chef_paths.rb

```diff
@@ -8,8 +8,8 @@ def initialize(app, env)
       
       def call(env)
         # there has got to be a better way
-        provisioners_list = env[:machine].config.vm.provisioners
-        chef_solo = provisioners_list.find { |p| p.name === :chef_solo }
+        key = ::Gem::Specification::find_by_name('vagrant').version < Gem::Version.new('1.7.0') ? :name : :type
```

Comment:
Is this necessary?  Could we just introduce a gem dependency constraint on vagrant > 1.7.0, and then have simpler code?